structs/ctl/message.go

```diff
@@ -125,3 +126,25 @@ func (r *ResponseWrapper) Encode() ([]byte, error) {
 
 	return msg, nil
 }
+
```

Comment:
`WebsoscketControl` came out of a problem I spent a little while thinking about. When theres an error in the key exchange process, we need to send an error message back to the client. However, regular HTTP status headers don't work over websocket, and the client is expecting to read the response message into the go `ctl` struct for that part of the exchange. If we sent an error message, we would get some sort of unmarshaling error, unless we used some type of reflection or message type identification to let the client know what it was decoding first. I think `WebsocketControl` turns out to be a much simpler solution, because its embedded in each message, so we have no issues figuring out what to read the message into when its received, and we just have to check whether the error value is set to see if anything went wrong, otherwise things are ok

Reply:
ah ok i see. i think the way we do it at work is have
```
type WebsocketMessage struct {
 messageType string
 body *json.RawMessage
}
```
and then depending on the message type you decide how to unmarshal json.